pages/dashboard/workspaces/[workspaceid]/collections/[collectionid]/settings/creators.vue

```diff
@@ -66,65 +76,131 @@ if (error.value) {
 }
 
 if (data.value) {
-  moduleData.creators = normalizeCreators(data.value);
+  creators.value = normalizeCreators(data.value);
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (code_refinement):** Ensure idempotency of `normalizeCreators` function.

Given that `normalizeCreators` is called multiple times with potentially the same data, ensure this function is idempotent or consider if it's necessary to call it multiple times.

```suggestion
// Ensure idempotency by checking if the data has changed before normalization
if (!deepEqual(creators.value, normalizeCreators(data.value))) {
  creators.value = normalizeCreators(data.value);
}
```

```